configuration.go

```diff
@@ -87,3 +107,106 @@ func (c *Client) GetConfiguration(applicationName string, profiles ...string) (S
 func joinProfiles(profiles []string) string {
 	return strings.Join(profiles, ",")
 }
+
+func (s *Source) toMap() {
+	result := map[string]interface{}{}
+	for _, propertySource := range s.PropertySources {
+		for key, value := range propertySource.Source {
+			entries := strings.Split(key, ".")
+			result = insertInMapRecursion(entries, value, result)
+		}
+	}
+	s.Data = result
+}
+
+func insertInMap(s []string, value interface{}, dest map[string]interface{}) map[string]interface{} {
+	keys := s[:len(s)-1]
+	last := s[len(s)-1]
+
+	curr := dest
+	for _, key := range keys {
+		switch curr[key].(type) {
+		case nil:
+			curr[key] = map[string]interface{}{}
+			curr = curr[key].(map[string]interface{})
+		case map[string]interface{}:
+			curr = curr[key].(map[string]interface{})
+		}
+	}
+	if curr[last] == nil {
+		curr[last] = value
+	}
+	return dest
+}
+
+func insertInMapRecursion(s []string, value interface{}, dest map[string]interface{}) map[string]interface{} {
+	key := s[0]
+	if len(s) > 1 {
+		switch dest[key].(type) {
+		case nil:
+			dest[key] = insertInMap(s[1:], value, map[string]interface{}{})
+		case map[string]interface{}:
+			dest[key] = insertInMap(s[1:], value, dest[key].(map[string]interface{}))
+		}
+	} else if len(s) == 1 {
+		if dest[key] == nil {
+			dest[key] = value
+		}
+	}
+	return dest
+}
+
+type BasicAuthTransport struct {
```

Comment:
this bit and down seems redundant and is more application specific implementation of this library. I do not think it belong in this library.